src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Logging/DocumentationSignatureGenerator.PartVisitor.cs

```diff
@@ -162,10 +162,11 @@ protected override void AppendNameForNamespaceType(StringBuilder sb, DefType typ
             protected override void AppendNameForInstantiatedType(StringBuilder builder, DefType type)
             {
                 int containingArity = 0;
-                DefType containingType = type.ContainingType;
+                TypeDesc containingType = InstantiateContainingType(type);
```

Comment:
It looks odd to have this API return TypeDesc. Should this return `DefType`?

Reply:
I copied it from the `TypeDesc.InstantiateSignature`, but I guess that one has to be a `TypeDesc`.
I must admit I'm still rather confused around the different types in the OM - specifically what operations will change the type. Probably my history with Cecil is at play here - since in Cecil a generic instantiation is a specific type which behaves more like a type reference.

I'll add a cast.

Reply:
It stood out to me because owning type of a nested type can only be a `DefType` (arrays, function pointers, etc. don't have nested types).

Having this as a DefType will produce a tiny (and likely meaningless in the grand scheme of things) improvement because we call a more specific `AppendName` overload below.

`InstantiateSignature` is a virtual method on `TypeDesc` because it's an operation valid on all types. E.g. `InstantiateSignature` on `!0[]` (array of the first generic parameter) would not return `DefType`. Now that we have support for covariant return types in C#, we could potentially tighten this so that if one is calling `InstantiateSignature` on a `DefType`, one gets a `DefType` back. Would be interesting to see how many casts that would get rid of. `InstantiateSignature` will always return the same type back, unless it's called on a signature variable itself.